src/Elm/Interface.elm

```diff
@@ -202,17 +201,16 @@ fileToDefinitions file =
                 )
                 file.declarations
     in
+    -- I really don't understand what this part of the function does.
     allDeclarations
-        |> List.map Tuple.first
-        |> List.Extra.unique
         |> List.map
-            (\x ->
-                ( x
-                , allDeclarations
-                    |> List.filter (Tuple.first >> (==) x)
-                )
+            (\( name, _ ) ->
+                -- AFAIK, it's not possible to have two declarations with the same name
```

Comment:
I mean, it is possible, it's just not valid Elm code. That said, it doesn't seem like it should be a syntax error so maybe elm-syntax should allow multiple declarations to appear?

Reply:
You're right. I'm not sure what to would be best here. In practice I don't think it matters at all. Shall we just close this PR? This is about simplifications/performance anyway, so not something super important.

Reply:
Sure we can revisit this if it ever become a problem in practice